storage/src/main/java/tech/pegasys/artemis/storage/CombinedChainDataClient.java

```diff
@@ -178,4 +185,49 @@ private boolean isFinalized(final UnsignedLong slot) {
     }
     return Optional.ofNullable(recentChainData.getBestBlockRoot());
   }
+
+  public boolean isStoreAvailable() {
+    return (recentChainData.getStore() != null);
+  }
+
+  /**
+   * Gets a list of CommitteeAssignments for a specified epoch.
+   *
+   * @param epoch - the current or historic epoch
+   * @return list of CommitteeAssignments
+   */
+  public SafeFuture<List<CommitteeAssignment>> getCommitteeAssignmentAtEpoch(UnsignedLong epoch) {
+    final UnsignedLong startingSlot = compute_start_slot_at_epoch(epoch);
+
+    SafeFuture<Optional<BeaconState>> future =
+        getStateAtSlot(startingSlot, recentChainData.getBestBlockRoot());
+    if (future != null) {
+      return future
+          .thenCompose(
+              optionalState ->
+                  completedFuture(getCommitteesFromStateWithCache(optionalState, startingSlot)))
```

Comment:
You can just use `thenApply` here and not need to wrap in a `completedFuture`.

Reply:
sounds less messy